# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: 3b777b313b76
# Full Hash: 3b777b313b76b517c66cba5278f06a2da9a6fb55
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 2
# Description:
#   Bug 1654112 - Update VideoConduit to use new APIs. r=ng
#   
#   This also disables the gtests until we figure out the new API.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D130065
# ==============================================================================

diff -r ae0715e32c27 -r 3b777b313b76 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Fri Oct 30 14:22:28 2020 -0400
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Fri Oct 30 14:30:28 2020 -0400
@@ -22,18 +22,13 @@
 #undef FF
 // Video Engine Includes
 #include "api/video_codecs/video_encoder_factory.h"
-#include "webrtc/call/call.h"
-#include "webrtc/common_types.h"
-#ifdef FF
-#  undef FF  // Avoid name collision between scoped_ptr.h and nsCRTGlue.h.
-#endif
-#include "webrtc/api/video_codecs/video_decoder.h"
-#include "webrtc/api/video_codecs/video_encoder.h"
-#include "webrtc/api/video_codecs/sdp_video_format.h"
-#include "webrtc/common_video/include/i420_buffer_pool.h"
-#include "webrtc/media/base/videosinkinterface.h"
-#include "webrtc/media/base/videoadapter.h"
-#include "webrtc/media/base/videobroadcaster.h"
+#include "api/video_codecs/video_decoder.h"
+#include "api/video_codecs/video_encoder.h"
+#include "api/video_codecs/sdp_video_format.h"
+#include "call/call.h"
+#include "common_video/include/i420_buffer_pool.h"
+#include "media/base/video_adapter.h"
+#include "media/base/video_broadcaster.h"
 #include <functional>
 #include <memory>
 /** This file hosts several structures identifying different aspects
@@ -69,7 +64,6 @@
  */
 class WebrtcVideoConduit
     : public VideoSessionConduit,
-      public webrtc::RtcpEventObserver,
       public webrtc::RtpPacketSinkInterface,
       public webrtc::Transport,
       public webrtc::VideoEncoderFactory,
@@ -316,10 +310,6 @@
     mRecvStreamStats.RecordTelemetry();
   }
 
-  void OnRtcpBye() override;
-
-  void OnRtcpTimeout() override;
-
   void SetRtcpEventObserver(mozilla::RtcpEventObserver* observer) override;
 
  private:
@@ -591,7 +581,7 @@
   int mMinBitrate = 0;
   int mStartBitrate = 0;
   int mPrefMaxBitrate = 0;
-  int mNegotiatedMaxBitrate = 0;
+  // int mNegotiatedMaxBitrate = 0;
   int mMinBitrateEstimate = 0;
 
   // Set to true to force denoising on.